### PR TITLE
Add integration script and workflow for tdx server

### PR DESCRIPTION
**Pull request by arronwy**
The integtation script is implemented by Wei.

Co-authored-by: Wei Liu <wei3.liu@intel.com>
Signed-off-by: Wei Liu <wei3.liu@intel.com>
Signed-off-by: Arron Wang <arron.wang@intel.com>

### CODE REVIEW

**Comment by jyao1** [comment]
> The integtation script is implemented by Wei.
> 
> Co-authored-by: Wei Liu [wei3.liu@intel.com](mailto:wei3.liu@intel.com) 

If it is implemented by Wei, I think we need: Signed-off-by: Wei Liu

**Comment by arronwy** [comment]
> > The integtation script is implemented by Wei.
> > Co-authored-by: Wei Liu [wei3.liu@intel.com](mailto:wei3.liu@intel.com)
> 
> If it is implemented by Wei, I think we need: Signed-off-by: Wei Liu Also, you may use multiple signed-off, if you modified it.

Yes, fixed in commit messages

File: .github/workflows/integration-tdx.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      - name: Build TdShim
+        uses: actions-rs/cargo@v1
+        with:
+          command: xbuild
+          args: -p td-shim --target x86_64-unknown-uefi --release --features=main,tdx
+
+      - name: Build PE format payload
+        run: |
+          cargo xbuild -p td-payload --target x86_64-unknown-uefi --release --features=main,tdx
+          cargo run -p td-shim-tools --features=td-shim/default --bin td-shim-ld -- target/x86_64-unknown-uefi/release/ResetVector.bin target/x86_64-unknown-uefi/release/td-shim.efi target/x86_64-unknown-uefi/release/td-payload.efi -o target/x86_64-unknown-uefi/release/final-pe.bin
```
Comment (fidencio) [inline .github/workflows/integration-tdx.yml:59]: @arronwy, if https://github.com/confidential-containers/td-shim/pull/223 gets merged before this one, I´d  consider changing these 2 lines to, instead, just doing
```
          bash sh_script/build_final.sh pe_boot_kernel
```

That would ensure that any changes that need to be done in the future, are only done as part of that script.

File: .github/workflows/integration-tdx.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+      - name: Run Tests
+        run: bash sh_script/integration_tdx.sh -f target/x86_64-unknown-uefi/release/final-pe.bin
```
Comment (fidencio) [inline .github/workflows/integration-tdx.yml:62]: Then it'd require changing the binary name here to `final-pe-boot-kernel.bin` instead.

**Review by fidencio — COMMENTED** [review]
@arronwy, I've made some suggestions as your work can benefit of a series that I submitted Yesterday.

Please, if possible, consider those.

File: .github/workflows/integration-tdx.yml (hunk 1 above)
Comment (arronwy) [reply .github/workflows/integration-tdx.yml:59]: good suggestion, we can use this script to avoid duplicate and reduce the maintenance effort.

Comment (arronwy) [reply .github/workflows/integration-tdx.yml]: @fidencio thanks, it works great!.

**Comment by jiangliu** [comment]
As we are switching from staging to main, should we target this PR against the main branch?

**Comment by jyao1** [comment]
> As we are switching from staging to main, should we target this PR against the main branch?

Yes. The patch will be applied to main as well.

**Review by jyao1 — APPROVED** [review]

**Comment by jyao1** [comment]
Drop this because staging is no longer needed.
Patch for main is at https://github.com/confidential-containers/td-shim/pull/237.

**Pull request closed without merging**